packages/traceability/issuer-irec-api/src/pods/certificate/handler/get-irec-certificates-to-import.handler.ts

```diff
@@ -0,0 +1,42 @@
+import { Repository } from 'typeorm';
+import { CommandHandler, ICommandHandler, QueryBus } from '@nestjs/cqrs';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { IssueWithStatus } from '@energyweb/issuer-irec-api-wrapper';
+import { GetAllCertificationRequestsQuery } from '@energyweb/issuer-api';
+import { IREC_SERVICE, IrecService } from '@energyweb/origin-organization-irec-api';
+
+import { GetIrecCertificatesToImportCommand } from '../command';
+import { IrecCertificate } from '../irec-certificate.entity';
+import { FullCertificationRequestDTO } from '../../certification-request';
+import { Inject } from '@nestjs/common';
+
+@CommandHandler(GetIrecCertificatesToImportCommand)
+export class GetIrecCertificatesToImportHandler
+    implements ICommandHandler<GetIrecCertificatesToImportCommand>
+{
+    constructor(
+        private readonly queryBus: QueryBus,
+        @InjectRepository(IrecCertificate)
+        private readonly repository: Repository<IrecCertificate>,
+        @Inject(IREC_SERVICE)
+        private readonly irecService: IrecService
+    ) {}
+
+    async execute({
+        user: { id, organizationId }
+    }: GetIrecCertificatesToImportCommand): Promise<IssueWithStatus[]> {
+        const issues = await this.irecService.getCertificates(id);
+        const certificationRequests = await this.queryBus.execute<
+            GetAllCertificationRequestsQuery,
+            FullCertificationRequestDTO[]
+        >(new GetAllCertificationRequestsQuery({ owner: String(organizationId) }));
+
+        return issues.filter((issue) => {
+            const certificationRequest = certificationRequests.find(
```

Comment:
This is super NIT, but there is `Array.prototype.some` method which returns boolean 🤷

Reply:
with .some() it loose its readability

Reply:
Up to you 👌